.github/workflows/enos-ci-bootstrap.yml

```diff
@@ -1,19 +1,26 @@
 name: enos-ci-bootstrap
 
 on:
+  workflow_dispatch:
   pull_request:
     branches:
       - main
   push:
     branches:
       - main
+      - mbaum/qt-318 # TODO: remove this after testing is complete
     paths:
       - enos/ci/**
       - .github/workflows/enos-bootstrap-ci.yml
 
 jobs:
   bootstrap-ci:
     runs-on: ubuntu-latest
+    env:
```

Comment:
This env change is good stuff.

Reply:
I've been thinking more about this and there are some strange things.

* We are triggering it twice. Once by the `pull_request` event and another by the `push` event.
* We should all the paths guards to the `pull_request` event trigger

Reply:
- The two triggers are required since the `pull_request` trigger does not run when the PR is merged and there is no option to enable that. So the way it works is that the `pull_request` trigger is run whenever a PR is opened, synchronized, or reopened.  Synchronized means a push to the PR has happened, this can be one or many commits. Regardless of how many commits are pushed to a PR in one push, the workflow will only run once. Once the PR is merged the workflow is run again due to the `push` trigger. Since the `push` trigger is limited to the main branch this covers the merge PR case not covered by the `pull_request` trigger. Another thing to note is that the workflow skips the `apply` step unless the branch is `main`. So therefore whenever a PR is opened and updated the workflow is run, but only up to the `plan` step. Once the PR is merged, the whole workflow runs, including `apply`

- I added the paths to the `pull_request` workflow trigger since the workflow was being run for all PRs, good catch.